keps/sig-multicluster/1645-multi-cluster-services-api/specification.md

```diff
@@ -0,0 +1,254 @@
+
+
+
+*   Record Format:
+    *   `<service>.<ns>.svc.<clustersetzone>. IN A <endpoint-ip>`
+*   Question Example
+    *   `myservice.test.svc.clusterset.local IN A`
+*   Answer Example:
+    *   `myservice.test.svc.clusterset.local 4 IN A 10.42.42.42`
+    *   `myservice.test.svc.clusterset.local 4 IN A 10.10.10.10`
+
+There must also be an `A` record of the following form for each ready endpoint with hostname of `<hostname>`, member cluster ID of `<clusterid>`, and IPv4 address `<endpoint-ip>`. If there are multiple IPv4 addresses for a given hostname, then there must be one such `A` record returned for each IP.
+
+
+
+*   Record Format:
+    *   `<hostname>.<clusterid>.<service>.<ns>.svc.<clustersetzone>. <ttl> IN A <endpoint-ip>`
+*   Question Example:
+    *   `my-hostname.721ab723-13bc-11e5-aec2-42010af0021e.myservice.test.svc.clusterset.local. IN A`
+*   Answer Example:
+    *   `my-hostname.721ab723-13bc-11e5-aec2-42010af0021e.myservice.test.svc.clusterset.local. 4 IN A 10.3.0.100`
+
+There must be an `AAAA` record for each _ready_ endpoint of the headless Service with IPv6 address `<endpoint-ip>` as shown below. If there are no _ready_ endpoints for the headless Service, the answer should be `NXDOMAIN`.
+
+
+
+*   Record Format:
+    *    `<service>.<ns>.svc.<clustersetzone>. <ttl> IN AAAA <endpoint-ip>`
+*   Question Example:
+    *    `headless.test.svc.clusterset.local. IN AAAA`
+*   Answer Example:
+    *    `headless.test.svc.clusterset.local. 4 IN AAAA 2001:db8::1`
+    *    `headless.test.svc.clusterset.local. 4 IN AAAA 2001:db8::2`
+    *    `headless.test.svc.clusterset.local. 4 IN AAAA 2001:db8::3`
+
+There must also be an `AAAA` record of the following form for each ready endpoint with hostname of `<hostname>`, member cluster ID of `<clusterid>`, and IPv6 address `<endpoint-ip>`. If there are multiple IPv6 addresses for a given hostname, then there must be one such `A` record returned for each IP.
+
+
+
+*   Record Format:
+    *   `<hostname>.<clusterid>.<service>.<ns>.svc.<clustersetzone>. <ttl> IN AAAA <endpoint-ip>`
+*   Question Example:
+    *   `my-hostname.721ab723-13bc-11e5-aec2-42010af0021e.test.svc.clusterset.local. IN AAAA`
+*   Answer Example:
+    *   `my-hostname.721ab723-13bc-11e5-aec2-42010af0021e.test.svc.clusterset.local. 4 IN AAAA 2001:db8::1`
+
+#### 2.4.2 - `SRV` Records
+
+For each combination of _ready_ endpoint with _hostname_ of `<hostname>`, member cluster ID of `<clusterid>`, and port in the Service with name `<port>` and number `<port-number>` using protocol `<proto>`, an `SRV` record of the following form must exist.
+
+
+
+*   Record Format:
+    *    `_<port>._<proto>.<service>.<ns>.svc.<clustersetzone>. <ttl> IN SRV <weight> <priority> <port-number> <hostname>.<clusterid>.<service>.<ns>.svc.<clustersetzone>.`
+
+This implies that if there are **N** _ready_ endpoints and the Service defines **M** named ports, there will be **N** X **M**  **`SRV`** RRs for the Service.
```

Comment:
This section doesn't read quite clearly to me. I think what you are saying is that the records returned should be the union of all the records for each clusterID. So there the _ready_ endpoints are across all clusters. I think it does say that, but it's a little ambiguous. Having an example with multiple cluster IDs may help clarify.

Also, using the UUID makes it a little hard to read the records and see for example that they are different clusters (if you were to add that example).

Reply:
I've cleaned up this example with multiple cluster IDs and changed up this and other examples to use easier-to-read cluster IDs in f992bc2.